File: domainObjects/src/main/java/com/szymkowski/personaltrainercompanion/trainings/domain/Training.java
package com.szymkowski.personaltrainercompanion.trainings.domain;

import com.j256.ormlite.field.DatabaseField;
import com.j256.ormlite.table.DatabaseTable;
import com.szymkowski.personaltrainercompanion.core.BaseEntity;

import org.joda.time.DateTime;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

@EqualsAndHashCode(callSuper = true)
@Data
@NoArgsConstructor
@AllArgsConstructor(suppressConstructorProperties = true)
@DatabaseTable(tableName = Training.TABLE_NAME)
class Training extends BaseEntity {
    public static final String TABLE_NAME = "table_trainings";

    @DatabaseField(generatedId = true)
    private Long id;

    public Training(DateTime trainingDate) {
        super(trainingDate);
        this.id = 0L;
    }

}

File: domainObjects/src/main/java/com/szymkowski/personaltrainercompanion/trainings/domain/Training.kt
package com.szymkowski.personaltrainercompanion.trainings.domain

import com.j256.ormlite.field.DatabaseField
import com.j256.ormlite.table.DatabaseTable
import com.szymkowski.personaltrainercompanion.core.CommonColumns
import org.joda.time.DateTime

@DatabaseTable(tableName = Training.TABLE_NAME)
data class Training (
        @DatabaseField(generatedId = true)
        val id : Long,

        @DatabaseField(columnName = CommonColumns.DATE_COLUMN)
        var date : DateTime

){



    constructor(trainingDate: DateTime) : this(0L, trainingDate)

    constructor() : this(0L, DateTime.now())

    companion object {
        const val TABLE_NAME = "table_trainings"
    }

}
